test/test_optim.py

```diff
@@ -700,5 +701,64 @@ def _test_reduce_lr_on_plateau(self, scheduler, targets, metrics, epochs=10, ver
                                            epoch, target[epoch], param_group['lr']), delta=1e-5)
 
 
+class TestEarlyStopping(TestCase):
+    def test_early_stopping_min_mode(self):
+        scores = [0.1, 0.2, 0.3, 0.2, 0.25]
+        targets = [True, True, True, False, False]
+        early_stopping = EarlyStoppingCriterion(patience=2, mode='min', min_delta=0.0)
+        for i, (s, t) in enumerate(zip(scores, targets)):
+            is_continue = early_stopping.step(s)
+            self.assertEqual(t, is_continue,
+                             message='Early stopping is wrong in epoch {}: expected {}, got {}'
+                             .format(i, t, is_continue))
+            if not t:
+                break
+
+    def test_early_stopping_max_mode(self):
+        scores = [0.3, 0.2, 0.1, 0.2, 0.25]
+        targets = [True, True, False, False, False]
+        early_stopping = EarlyStoppingCriterion(patience=1, mode='max', min_delta=0.0)
+        for i, (s, t) in enumerate(zip(scores, targets)):
+            is_continue = early_stopping.step(s)
+            self.assertEqual(t, is_continue,
+                             message='Early stopping is wrong in epoch {}: expected {}, got {}'
+                             .format(i, t, is_continue))
+            if not t:
+                break
+
+    def test_early_stopping_min_delta_1(self):
```

Comment:
test_early_stopping_min_mode, test_early_stopping_max_mode, test_early_stopping_min_delta_2 functions look the same, except of arguments: scores, targets, patience, mode, min_delta. I think it'll be better to create a single function instead.